### ISSUE TITLE
Migrate from express to something lighter

### ISSUE THREAD

**Issue by MarkMcCulloh** [winglang/wing#4506]
express is used in the simulator for our various resources that need to host an http server.

express works but it's:
- Has many dependencies, especially ones that are no longer needed in modern js
  - This can cause issues with peer dependencies as we move forward, especially due to our usage of bundled dependencies. e.g. express depends on v1 of mime but another lib we use depends on mime v2
- Not very well maintained anymore, hasn't been updated in over a year at the time of this post. (missing many dependency updates)
- Heavy handed, our use cases are generally pretty simplistic. We just need an HTTP server with routing

Alternatives to consider:
- https://github.com/honojs/hono
- https://github.com/koajs/koa
- use node builtin node HTTP server and handle routing ourselves

**Comment by ekeren** [comment]
I think it qualifies as a good first issue
